core/src/smartcontracts/isi/mod.rs

```diff
@@ -722,4 +722,57 @@ mod tests {
         );
         Ok(())
     }
+
+    #[test]
+    fn executing_unregistered_trigger_should_return_error() -> Result<()> {
+        let wsv = WorldStateView::new(world_with_test_domains()?);
+        let account_id = AccountId::from_str("alice@wonderland")?;
+        let trigger_id = TriggerId::from_str("test_trigger_id")?;
+
+        assert!(matches!(
+            ExecuteTriggerBox::new(trigger_id)
+                .execute(account_id, &wsv)
+                .expect_err("Error expected"),
+            Error::Find(_)
+        ));
+
+        Ok(())
+    }
+
+    #[test]
+    fn unauthorized_trigger_execution_should_return_error() -> Result<()> {
+        let wsv = WorldStateView::new(world_with_test_domains()?);
+        let account_id = AccountId::from_str("alice@wonderland")?;
+        let fake_account_id = AccountId::from_str("fake@wonderland")?;
```

Comment:
It would be nice to register this fake account for a correct test.

```rust
let register_account = RegisterBox::new(Account::new(
        AccountId::new(fake_account, ...
```

Reply:
Thanks!